### ISSUE TITLE
CSRF verification failed when trying to create new options from Select2

### ISSUE THREAD

**Issue by borfast** [yourlabs/django-autocomplete-light#772]
I have a Select2 field on a ModelForm. This field is supposed to allow users to add new options but I always get `CSRF verification failed. Request aborted.` from the view that was supposed to help create the new object, and I get `Forbidden (CSRF token missing or incorrect.): /movement-autocomplete/` in the console logs.

Inspecting the HTTP POST request, I can see that its "x-csrftoken" header is set to "null", which I suppose explains the error.

This only happens when DEBUG is False, otherwise it works perfectly.

Is this a known issue that turns out to be something subtle that people miss? Any ideas on what could be wrong?

I'm setting up the field like this:

```Python
widgets = {
    'movement': autocomplete.ModelSelect2(
        url='movement-autocomplete',
        attrs={'data-placeholder': 'Movement'}
    ),
}
```

The view that feeds the field is the following:

```Python
@method_decorator(login_required, name='dispatch')
class MovementAutocomplete(autocomplete.Select2QuerySetView):
    create_field = 'name'

    def has_add_permission(self, request):
        """Allow any signed in user to add a new movement."""
        if not request.user.is_authenticated():
            return False

        return True

    def get_queryset(self):
        qs = Movement.objects.all()

        if self.q:
            qs = qs.filter(name__istartswith=self.q)

        return qs
```

**Comment by borfast** [comment]
Figured it out. Apparently jQuery isn't sending the headers as it should but  it's an easy enough fix. I'll submit a pull request for it.

**Comment by borfast** [comment]
Like I said in my pull request, I don't drink but if I did, I'd say I was probably drunk when I looked at this.

Turns out the problem was simply the fact that I had CSRF_COOKIE_HTTPONLY set to True, which of course means the CSRF cookie can't be read from Javascript.

In this case the only option is to fetch the cookie from the DOM, as mentioned [in the docs](https://docs.djangoproject.com/en/1.10/ref/settings/#std:setting-CSRF_COOKIE_HTTPONLY).

Perhaps DAL could be changed to look at this setting and fetch the cookie from the right place.
